api/v1alpha1/customnbimage_types.go

```diff
@@ -212,3 +224,22 @@ func (r *CustomNBImageRuntimeSpec) isValid() bool {
 
 	return true
 }
+
+// hasValidImagePullSecret checks if the ImagePullSecret is valid, eg name is not empty
+func (b *BuildTypeSpec) hasValidImagePullSecret() bool {
+	if b.ImagePullSecret != (ImagePullSecret{}) {
+		if b.ImagePullSecret.Name == "" {
+			return false
+		}
+	}
+
+	return true
+}
```

Comment:
Should that not be the job of an admission controller instead ?

Reply:
It should be checked during reconciliation, as the secret must not exist at CNBi resource creating time. see [#111](https://github.com/thoth-station/meteor-operator/issues/111)